modules/bag/tests/test_proto_conversion.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <random>
+
+#include "hephaestus/random/random_container.h"
+#include "hephaestus/serdes/protobuf/concepts.h"
+#include "test.pb.h"
+
+namespace heph::bag::tests {
+struct User {
+  auto operator==(const User& other) const -> bool = default;
+  [[nodiscard]] static auto random(std::mt19937_64& mt) -> User;
+
+  std::string name;
+  int age{};
+  std::vector<float> scores;
+};
+
+auto User::random(std::mt19937_64& mt) -> User {
+  return { .name = random::randomT<std::string>(mt),
+           .age = random::randomT<int>(mt),
+           .scores = random::randomT<std::vector<float>>(mt) };
+}
+
+struct Company {
+  auto operator==(const Company& other) const -> bool = default;
+  [[nodiscard]] static auto random(std::mt19937_64& mt) -> Company;
+
+  std::string name;
+  int employer_count{};
+};
+
+auto Company::random(std::mt19937_64& mt) -> Company {
+  return { .name = random::randomT<std::string>(mt), .employer_count = random::randomT<int>(mt) };
+}
```

Comment:
nit: kinda weird examples for the test in  a robotics scenario :)

Reply:
good point, renamed